paddle/fluid/inference/analysis/analyzer.cc

```diff
@@ -101,7 +101,11 @@ Analyzer::Analyzer() { Register("manager1", new DfgPassManagerImpl); }
 
 void Analyzer::Run(Argument* argument) {
   std::vector<std::string> passes;
-  for (auto& pass : all_ir_passes_) {
+  if (use_mkldnn_) {
+    VLOG(3) << "Adding MKL-DNN placement pass";
+    passes.push_back("mkldnn_placement_pass");
+  }
```

Comment:
You can add `#ifdef PADDLE_WITH_MKLDNN` in next PR to ensure that `use_mkldnn_=true` only in `PADDLE_WITH_MKLDNN`.
```
#ifdef PADDLE_WITH_MKLDNN
if (use_mkldnn_) {
    VLOG(3) << "Adding MKL-DNN placement pass";
    passes.push_back("mkldnn_placement_pass");
  }
#endif
```